instructionAPI/src/AMDGPU/gfx908/finalizeOperands.C

```diff
@@ -1,2892 +1,2265 @@
+                break;
+                case 65:case 96:case 98:case 99:case 100:case 101:case 102:case 103:case 104:case 105:case 106:case 107:case 108:
+                //GLOBAL_ATOMIC_CMPSWAP,GLOBAL_ATOMIC_SWAP_X2,GLOBAL_ATOMIC_ADD_X2,GLOBAL_ATOMIC_SUB_X2,GLOBAL_ATOMIC_SMIN_X2,GLOBAL_ATOMIC_UMIN_X2,GLOBAL_ATOMIC_SMAX_X2,GLOBAL_ATOMIC_UMAX_X2,GLOBAL_ATOMIC_AND_X2,GLOBAL_ATOMIC_OR_X2,GLOBAL_ATOMIC_XOR_X2,GLOBAL_ATOMIC_INC_X2,GLOBAL_ATOMIC_DEC_X2
+                appendOPR_VGPR(layout.VDST,false,true,2);
+                if(layout.SADDR == 0x7f)
+                appendOPR_VGPR(layout.ADDR,true,false,2);
+                else
+                appendOPR_VGPR(layout.ADDR,true,false);
+                appendOPR_VGPR(layout.DATA,true,false,2);
+                if(layout.SADDR != 0x7f){
+                    appendOPR_SREG(layout.SADDR,true,false,2);
+                }
+                appendOPR_SDST_M0(124,true,false,1,true);
+                insn_in_progress->appendOperand(makeRegisterExpression(amdgpu_gfx908::vmcnt,0,36),true,true,true);
+                insn_in_progress->appendOperand(makeRegisterExpression(amdgpu_gfx908::lgkmcnt,0,16),true,true,true);
+                break;
+                case 97:
+                //GLOBAL_ATOMIC_CMPSWAP_X2
+                appendOPR_VGPR(layout.VDST,false,true,4);
+                if(layout.SADDR == 0x7f)
+                appendOPR_VGPR(layout.ADDR,true,false,2);
+                else
+                appendOPR_VGPR(layout.ADDR,true,false);
+                appendOPR_VGPR(layout.DATA,true,false,4);
+                if(layout.SADDR != 0x7f){
+                    appendOPR_SREG(layout.SADDR,true,false,2);
+                }
+                appendOPR_SDST_M0(124,true,false,1,true);
+                insn_in_progress->appendOperand(makeRegisterExpression(amdgpu_gfx908::vmcnt,0,36),true,true,true);
+                insn_in_progress->appendOperand(makeRegisterExpression(amdgpu_gfx908::lgkmcnt,0,16),true,true,true);
+                break;
+            }
+        }
+
+        void InstructionDecoder_amdgpu_gfx908::finalizeENC_FLAT_SCRATCHOperands(){
+            layout_ENC_FLAT_SCRATCH & layout = insn_layout.ENC_FLAT_SCRATCH;
+            switch (layout.OP) {
+                case 16:case 17:case 18:case 19:case 20:case 32:case 33:case 34:case 35:case 36:case 37:
+                //SCRATCH_LOAD_UBYTE,SCRATCH_LOAD_SBYTE,SCRATCH_LOAD_USHORT,SCRATCH_LOAD_SSHORT,SCRATCH_LOAD_DWORD,SCRATCH_LOAD_UBYTE_D16,SCRATCH_LOAD_UBYTE_D16_HI,SCRATCH_LOAD_SBYTE_D16,SCRATCH_LOAD_SBYTE_D16_HI,SCRATCH_LOAD_SHORT_D16,SCRATCH_LOAD_SHORT_D16_HI
+                appendOPR_VGPR(layout.VDST,false,true);
+                if(layout.SADDR == 0x7f)
+                appendOPR_VGPR(layout.ADDR,true,false,2);
+                else
+                appendOPR_VGPR(layout.ADDR,true,false);
+                if(layout.SADDR != 0x7f)
+                appendOPR_SREG(layout.SADDR,true,false,2);
```

Comment:
indent! This is difficult to read.  Prefer using a new block instead of a single statement:

```
                if (layout.SADDR == 0x7f)  {
                    appendOPR_VGPR(layout.ADDR,true,false,2);
                }  else  {
                    appendOPR_VGPR(layout.ADDR,true,false);
                }
                if (layout.SADDR != 0x7f)  {
                    appendOPR_SREG(layout.SADDR,true,false,2);
                }
```